controlplane/kubeadm/api/v1beta1/kubeadm_control_plane_types.go

```diff
@@ -223,6 +280,25 @@ type KubeadmControlPlaneStatus struct {
 	// Conditions defines current service state of the KubeadmControlPlane.
 	// +optional
 	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+
+	// LastRemediation stores info about last remediation performed.
+	// +optional
+	LastRemediation *LastRemediationStatus `json:"lastRemediation,omitempty"`
+}
+
+// LastRemediationStatus  stores info about last remediation performed.
+// NOTE: if for any reason information about last remediation are lost, RetryCount is going to restart from 0 and thus
+// more remediations than expected might happen.
+type LastRemediationStatus struct {
+	// Machine is the machine name of the latest machine being remediated.
+	Machine string `json:"machine"`
```

Comment:
Instead of storing this information in status, could we track it on the Machine themselves via annotations?

Reply:
@vincepri How do we decide when information should end up in status vs annotations?

Reply:
I think we can compute the last remediated machine from the list of machines, but not the last remediation time nor the number of retries, unless we found a way to trickle down that information from the first part of the remediation (delete) to the second part (create the replacement machine), e.g. expanding the scope of RemediationInProgressAnnotation. opinions?

Reply:
Storing it on the individual Machines sounds better to me

Reply:
Q: For which information do we want to do this?

I assume that we don't want to hand over the "last remediated machine name" after delete to the create in the next reconcile? (as the Machin we would store that one is gone after the delete)

Reply:
Fixed, now last remediation data is stored at machine level and used by KCP as a source of truth